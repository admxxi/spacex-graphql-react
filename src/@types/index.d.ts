/* eslint-disable */
export interface IPost {
  id: number
  title: string
  body: string
}

export type Country = {
  code: !number
  name: !string
  emoji: !string
}

export type Mission = {
  mission_name: string
  // id: number
  // name: string
}
// export type Mission = {
//   description: string
//   id: ID
//   manufacturers: [string]
//   name: string
//   twitter: string
//   website: string
//   wikipedia: string
// }
