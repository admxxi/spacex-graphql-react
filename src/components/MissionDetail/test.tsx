import React from 'react'
import { render } from '@testing-library/react'
import { MissionDetail } from './index'
import { Launch as Ilaunch } from '../../@types/graphql'

const MockLaunch: Ilaunch = {
  __typename: 'Launch',
  id: '9999999',
  is_tentative: false,
  upcoming: false,
  mission_name: 'Dummy Mission',
  links: {
    __typename: 'LaunchLinks',
    article_link: 'https://dummy.article.com/',
    video_link: 'https://dummy.video.com/',
    flickr_images: [
      'https://live.staticflickr.com/65535/50618376646_8f52c31fc4_o.jpg'
    ],
    mission_patch: 'https://i.imgur.com/t5R4BAQ.png'
  },
  launch_date_utc: '2020-11-16T00:27:00.000Z',
  details: 'dummy details'
}

it('renders correctly', () => {
  const { container } = render(<MissionDetail launch={MockLaunch} />)
  expect(container.firstChild).toMatchSnapshot()
})
