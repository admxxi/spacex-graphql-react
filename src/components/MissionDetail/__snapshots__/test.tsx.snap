// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  class="MissionDetail__MissionDetailStyled-pzpvdl-0 gQgBpR card d-inline-block"
>
  <div
    class="card-image"
  >
    <figure
      class="image is-5by3"
    >
      <img
        alt="Placeholder image"
        height="auto"
        src="https://live.staticflickr.com/65535/50618376646_8f52c31fc4_o.jpg"
        width="100%"
      />
    </figure>
  </div>
  <div
    class="card-content"
  >
    <div
      class="media is-align-items-center"
    >
      <div
        class="media-left"
      >
        <figure
          class="image is-48x48"
        >
          <img
            alt="Placeholder image"
            height="auto"
            src="https://i.imgur.com/t5R4BAQ.png"
            width="100%"
          />
        </figure>
      </div>
      <div
        class="media-content"
      >
        <p
          class="ma-0 title is-4"
        >
          Dummy Mission
        </p>
      </div>
    </div>
    <div
      class="content"
    >
      <p>
        dummy details
      </p>
    </div>
    <p
      class="buttons"
    >
      <button
        class="button is-large is-link is-light is-info is-2"
        type="button"
      >
        <span
          class="icon"
        >
          <i
            class="fa fa-youtube"
          />
        </span>
        <span>
          Watch Video
        </span>
      </button>
      <button
        class="button is-large is-link is-light is-info is-2"
        type="button"
      >
        <span
          class="icon"
        >
          <i
            class="fa fa-wordpress"
          />
        </span>
        <span>
          See More
        </span>
      </button>
    </p>
  </div>
</div>
`;
